### PR TITLE
Update login/registration UI to match designs

### LINKED ISSUE #522: Update login/registration UI to match designs
**DEPENDS ON https://github.com/mitodl/bootcamp-ecommerce/issues/521**

[Designs](https://projects.invisionapp.com/d/main#/projects/prototypes/19928472) (the issue deals with the "create account" pages)

#### Acceptance criteria
- Captcha added to the appropriate forms
- All account creation pages updated to match the designs
- Login page styling updates to match the look and feel laid out in the designs (these pages aren't in the designs, but there should be enough in the other designs to make the desired styles clear)

### LINKED ISSUE #564: AttributeError: 'Settings' object has no attribute 'PASSWORD_RESET_CONFIRM_URL'
Sentry Issue: [BOOTCAMP-AQ](https://sentry.io/organizations/mit-office-of-digital-learning/issues/1688654153/?referrer=github_integration)

```
AttributeError: 'Settings' object has no attribute 'PASSWORD_RESET_CONFIRM_URL'
(12 additional frame(s) were not displayed)
...
  File "djoser/views.py", line 206, in send_password_reset_email
    settings.EMAIL.password_reset(self.request, context).send(to)
  File "templated_mail/mail.py", line 68, in send
    self.render()
  File "templated_mail/mail.py", line 60, in render
    context = make_context(self.get_context_data(), request=self.request)
  File "djoser/email.py", line 35, in get_context_data
    context['url'] = settings.PASSWORD_RESET_CONFIRM_URL.format(**context)
  File "django/utils/functional.py", line 257, in inner
    return func(self._wrapped, *args)
```

### PR DESCRIPTION
**Pull request by mbertrand**
Blocked by #521

#### What are the relevant tickets?
Closes #522
Closes #564

#### What's this PR do?
- copies NotificationContainer from xpro, enables it, gets it working
- moves `userNotifications` state inside `ui` state
- some styling and content changes
- fixes the forgot password functionality by adding djoser settings

#### How should this be manually tested?
- Try to hit all the login, registration, and profile pages and see if they look like they're supposed to and still work like they're supposed to.
- Go to `../forgot-password`, submit the form, click the link in the email you should get, and submit that form as well.  Your password should be successfully changed.

#### Any background context you want to provide?
I did not change the default size of the recaptcha to be larger than its native size, but did center it. 

#### Screenshots (if appropriate)

Screen Shot 2020-05-21 at 4 21 25 PM

-------------

Screen Shot 2020-05-21 at 4 21 45 PM

-------------

Screen Shot 2020-05-26 at 8 19 17 AM

-------------

Screen Shot 2020-05-26 at 8 21 05 AM

-------------

Screen Shot 2020-05-26 at 8 21 37 AM

-------------

Screen Shot 2020-05-21 at 4 33 15 PM

-------------

Screen Shot 2020-05-21 at 4 33 04 PM

--------------

Screen Shot 2020-05-26 at 8 59 38 AM

----------------

Screen Shot 2020-05-26 at 8 59 51 AM

-----------------

Screen Shot 2020-05-26 at 9 06 55 AM

-----------------

Screen Shot 2020-05-26 at 9 44 47 AM

### CODE REVIEW

**Comment by mbertrand** [comment]
@pdpinch @abdulkdawson screenshots are in the PR description

**Comment by pdpinch** [comment]
@abdulkdawson can you take a look at the screenshots?

**Comment by abdulkdawson** [comment]
This all looks great to me. @mbertrand where is the screenshot for step 1 of 2 in the create account flow? Also, The line spacing on the profile page needs to be increased and the profile labels (Year or Birth, Gender, Address etc) should be #7C7B7D.

**Comment by pdpinch** [comment]
I'm not sure if this is out of scope, but I noticed a couple of issues with the verification email:

image

**Comment by mbertrand** [comment]
@abdulkdawson I updated the description with the correct screenshot for step 1 of 2 for account creation, here it is again separately:

Screen Shot 2020-05-26 at 8 21 05 AM

**Comment by mbertrand** [comment]
@abdulkdawson seems like the current View/Edit profile pages are going to be replaced with drawers instead (#547) so I will apply the profile styling changes in that PR instead of this one if that's okay with you.

**Comment by mbertrand** [comment]
@pdpinch what should the logo be on the verification email? Same as on the new header  (MIT | Bootcamps)?

**Comment by abdulkdawson** [comment]
@mbertrand The 1 of 2 view looks great. Thanks.

**Comment by mbertrand** [comment]
@pdpinch here's the updated email template:

Screen Shot 2020-05-27 at 12 53 15 PM

File: mail/templates/email_base.html [hunk 1]
```diff
@@ -239,7 +239,7 @@
 	            <tr>
 	                <td style="padding: 20px;">
                       {% block logo %}
-                      <img src="{{ base_url }}/static/images/mitx-pro-logo.png" width="160" height="80" alt="{{ site_name }}" style="height: auto; background: #ffffff; font-family: sans-serif; font-size: 15px; line-height: 15px; color: #A31F34; margin-bottom: -10px;  margin-right: 10px;"/>
+                      <img src="{{ base_url }}/static/images/bootcamp-logo-black.png" width="160" height="80" alt="{{ site_name }}" style="height: auto; background: #ffffff; font-family: sans-serif; font-size: 15px; line-height: 15px; color: #A31F34; margin-bottom: -10px;  margin-right: 10px;"/>
```
[the remark this answers is not in the document]
Comment (mbertrand) [reply mail/templates/email_base.html:242]: That doesn't work - turns into a url of `http:///static/images/bootcamp-logo-black.png` in the email

File: main/templates/navbar.html [hunk 2]
```diff
@@ -4,14 +4,22 @@
...
           </div>
         <div class="pull-right">
           {% if authenticated %}
-          <a class="header-dashboard-link" href="/logout">
+          <a class="header-dashboard-link" href={%  url "logout" %}>
             Sign Out
           </a>
           {% endif %}
+          {% if social_auth_enabled and not authenticated %}
+          <a class="header-dashboard-link reverse" href={% url "login" %}>
```
Comment (noisecapella) [inline main/templates/navbar.html:16]: Can we add quotes around the `href` attributes in this file?

Comment (mbertrand) [reply main/templates/navbar.html]: oops, fixed

**Pull request merged**